Build failure: pkgsLLVM.jemalloc

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `pkgsLLVM.jemalloc`

### Build log

```
>>> referenced by lib/libjemalloc.so
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:534: test/integration/xallocx] Error 1
make: *** [Makefile:534: test/integration/slab_sizes] Error 1
aarch64-unknown-linux-gnu-ld: error: undefined reference due to --no-allow-shlib-undefined: std::__1::mutex::lock()
>>> referenced by lib/libjemalloc.so

aarch64-unknown-linux-gnu-ld: error: undefined reference due to --no-allow-shlib-undefined: std::set_new_handler(void (*)())
>>> referenced by lib/libjemalloc.so

aarch64-unknown-linux-gnu-ld: error: undefined reference due to --no-allow-shlib-undefined: std::__1::mutex::unlock()
>>> referenced by lib/libjemalloc.so

aarch64-unknown-linux-gnu-ld: error: undefined reference due to --no-allow-shlib-undefined: __cxa_begin_catch
>>> referenced by lib/libjemalloc.so

aarch64-unknown-linux-gnu-ld: error: undefined reference due to --no-allow-shlib-undefined: __cxa_end_catch
>>> referenced by lib/libjemalloc.so

aarch64-unknown-linux-gnu-ld: error: undefined reference due to --no-allow-shlib-undefined: std::__throw_bad_alloc()
>>> referenced by lib/libjemalloc.so

aarch64-unknown-linux-gnu-ld: error: undefined reference due to --no-allow-shlib-undefined: std::terminate()
>>> referenced by lib/libjemalloc.so

aarch64-unknown-linux-gnu-ld: error: undefined reference due to --no-allow-shlib-undefined: typeinfo for std::bad_alloc
>>> referenced by lib/libjemalloc.so

aarch64-unknown-linux-gnu-ld: error: undefined reference due to --no-allow-shlib-undefined: __gxx_personality_v0
>>> referenced by lib/libjemalloc.so
make: *** [Makefile:534: test/integration/sdallocx] Error 1
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:534: test/integration/thread_arena] Error 1
make: *** [Makefile:534: test/integration/thread_tcache_enabled] Error 1
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:534: test/integration/overflow] Error 1
make: *** [Makefile:534: test/integration/rallocx] Error 1
```

### Additional context

Add any other context about the problem here.

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@yu-re-ka 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"aarch64-linux"`
 - host os: `Linux 5.15.156, NixOS, 24.05 (Uakari), 24.05.20240418.bc279bb`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/run/current-system/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/libraries/jemalloc/default.nix
+++ b/pkgs/development/libraries/jemalloc/default.nix
@@ -26,6 +26,12 @@ stdenv.mkDerivation rec {
       url = "https://github.com/jemalloc/jemalloc/commit/d00ecee6a8dfa90afcb1bbc0858985c17bef6559.patch";
       hash = "sha256-N5i4IxGJ4SSAgFiq5oGRnrNeegdk2flw9Sh2mP0yl4c=";
     })
+    # fix linking with libc++, can be removed in the next update (after 5.3.0).
+    # https://github.com/jemalloc/jemalloc/pull/2348
+    (fetchpatch {
+      url = "https://github.com/jemalloc/jemalloc/commit/4422f88d17404944a312825a1aec96cd9dc6c165.patch";
+      hash = "sha256-dunkE7XHzltn5bOb/rSHqzpRniAFuGubBStJeCxh0xo=";
+    })
   ];
 
   configureFlags =

